File: LeetCode Problems/201 to 300/283. Move Zeroes/283. Move Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int> &nums) {
        int n = nums.size(), cnt = 0;
        vector<int> ans;
        for (int i = 0; i < n; i++) {
            if (nums[i] == 0)
                cnt++;
            else
                ans.push_back(nums[i]);
        }
        for (int i = 0; i < cnt; i++)
            ans.push_back(0);
        nums = ans;
    }
};

File: LeetCode Problems/201 to 300/283. Move Zeroes/283. Move Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int k = 0;

        // checking for the non zero elements
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                nums[k++] = nums[i];
            }
        }

        // remaining elements will be zero
        for (int i = k; i < nums.length; i++) {
            nums[i] = 0;
        }
    }
}